bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c

```diff
@@ -607,6 +607,7 @@ int pubsub_tcpHandler_listen(pubsub_tcpHandler_t *handle, char *url) {
             }
             celixThreadRwlock_unlock(&handle->dbLock);
         } else {
+            free(pUrl);
```

Comment:
correct, added close of fd.

Reply:
Then the fd leakage in `else` branch is fixed, while the leakage in the `if` branch remains, e.g., when `rc = listen(fd, SOMAXCONN);` failed.

I have improved the error injection technique mentioned in #399 so that it can be deployed easily in CMake projects.
To be honest, the one in #399 is ugly and cumbersome to use. Once we have such mechanism in place, emulating socket listening error is fairly easy, and code coverage will not be a problem. I plan to upstream it when working on #441.